### PR TITLE
[docs] - Add deployment guides to sidenav + missing guides page

### PR DESCRIPTION
**Pull request by erinkcochran87**
## Summary & Motivation

This PR:

- Adds the OSS deployment guides to the sidenav, which will now have them show up correctly in the sitemap
- Adds a category page for `/deployment/guides/kubernetes`, which was previously a 404

## How I Tested These Changes

👀

### CODE REVIEW

File: docs/content/deployment/guides/kubernetes.mdx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+title: "Dagster + Helm deployment guides | Dagster Docs"
+description: "Guides for deploying Dagster with Helm."
+---
+
+# Dagster + Helm deployment guides
+
+Check out these guides to learn the basics of Dagster deployment using Helm, including setting up Dagster and running the Dagster webserver. [Go back to all deployment guides](/deployment/guides).
```
Comment (gibsondan) [inline docs/content/deployment/guides/kubernetes.mdx:8]: I think we can take out "including setting up Dagster and running the Dagster webserver." (there are more things besides the webserver, like the daemon /database/ etc.)

**Review by gibsondan — APPROVED** [review]
sorry for delay

File: docs/content/_navigation.json [hunk 2]
```diff
@@ -357,7 +357,59 @@
...
+                    "path": "/deployment/guides/kubernetes/how-to-migrate-your-instance"
+                  }
+                ]
+              },
+              {
+                "title": "Deploying to Docker",
+                "path": "/deployment/guides/docker"
+              },
+              {
+                "title": "Deploying to Amazon Web Services EC2/ECS",
```
[the remark this answers is not in the document]
Comment (erinkcochran87) [reply docs/content/_navigation.json:397]: I had that thought and didn't know if we needed to be more granular here. We can always keep it in the page title though.

File: docs/content/deployment/guides/kubernetes.mdx (hunk 1 above)
Comment (erinkcochran87) [reply docs/content/deployment/guides/kubernetes.mdx:8]: True true. I'll take it out!

**Pull request merged**